### DISCUSSION TITLE
Ansible-lint check all files in repo, not only files in commit

### DISCUSSION THREAD

**Discussion by ingtarius** [ansible/ansible-lint#1121] [Q&A] [answered]
##### Summary
I try to add linters in big project, and i dont want to fix all errors in 1000+ files. Step-by-step, each commit chould be fixed with linters

##### Issue Type
- Bug Report

##### Ansible and Ansible Lint details
```console (paste below)
➜  ansible_linter_check git:(main) ansible --version
ansible 2.10.3
  config file = None
  configured module search path = ['/Users/user1/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.10.4/libexec/lib/python3.9/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.9.1 (default, Dec 10 2020, 10:36:35) [Clang 12.0.0 (clang-1200.0.32.27)]
➜  ansible_linter_check git:(main) pre-commit --version
pre-commit 2.9.3
➜  ansible_linter_check git:(main) ansible-lint --version
ansible-lint 4.3.7
```
- ansible installation method: one of source, pip, OS package
brew
- ansible-lint installation method: one of source, pip, OS package
brew

##### STEPS TO REPRODUCE
Create 2 roles with errors, push first role with skip pre-commit, try to commit second and see errors from firsr
I create simple repo with instruction: https://github.com/ingtarius/ansible_linter_check

##### Desired Behaviour
Linter check only second role, only files in commit

##### Actual Behaviour
Linter check all files in repo (git ls-files)

**Comment by ssbarnea** [comment]
Please use discussions if you have questions, not the issue tracker.

The linter is a complex piece of software, not a simple code style checker and a modification made to one file may produce an error in another file that was not touched by last commit.

Read the documentation about `--progressive` mode and that will likely make it easier for you to slowly adopt it the linter.

**Comment by ingtarius** [comment, marked as answer]
I solve this by using this entry in pre-commit
```
entry: bash -c "ansible-lint $(git diff --name-only --cached | egrep '.(yml|yaml)')"
```
This commit change policy, so i dont know best solution: https://github.com/ansible-community/ansible-lint/pull/932/files
